### PR TITLE
xConf update

### PR DESCRIPTION
**Pull request by Nipodemos**
### Whats new:
- now it gives support for you to use quotes on item name.
- if someone don't use quotes on an item that needs it, show a warning
- item that have slots or options didn't work , now it does.

### Tests:
i've used the following commands: 
```
do iconf Diamond Ring 0 0 1
do iconf Diver Goggles 0 0 1
do iconf Eye Patch 0 0 1
do iconf Flower Ring 0 0 1
do iconf Flu Mask 0 0 1
do iconf Full Plate 0 0 1
do iconf Gemmed Sallet 0 0 1
do iconf "Hood [1]" 0 0 1
do iconf 505 0 1 0 # Blue Potion
do iconf 545 50 1 0 # Condensed Red Potion
do iconf 547 50 1 0 # Condensed White Potion
do iconf 546 50 1 0 # Condensed Yellow Potion
do iconf 506 0 1 0 # Green Potion
do iconf 501 25 1 0 # Red Potion
do iconf 502 25 1 0 # Orange Potion
do iconf 504 25 1 0 # White Potion
do iconf 503 25 1 0 # Yellow potion
```

#### Output when first configures:
image

#### if you have already configured the items on items_control.txt:
image

#### tested mconf and it works fine

### CODE REVIEW

File: plugins/xConf/xConf.pl [hunk 1]
```diff
@@ -38,102 +38,121 @@ package xConf;
...
 	message "xConf plugin reloading or unloading\n", 'success'
 }
 
 sub xConf {
-my ($cmd, $args) = @_;
-	my ($file,$file2,$found,$key,$oldval,$shopname,$type,$value, $name, $inf_hash, $ctrl_hash);
+	my ($cmd, $args) = @_;
+	my ($file,$tables_file,$found,$key,$oldval,$shopname,$type,$value, $name, $inf_hash, $ctrl_hash, $id, $needQuotes);
+	
+	($key, $value) = $args =~ /([\s\S]+?)\s([\-\d\.]+[\s\S]*)/ if !$shopname;
```
Comment (lututui) [inline plugins/xConf/xConf.pl]: At this point, $shopname is always undef

File: plugins/xConf/xConf.pl [hunk 2]
```diff
@@ -38,102 +38,121 @@ package xConf;
...
 		$type = 'Item';
-	} elsif ($cmd eq 'mconf') {
-		$inf_hash  = \%monsters_lut;
-		$ctrl_hash = \%mon_control;
-		$file = 'mon_control.txt';
-		$file2 = 'tables\..\monsters.txt';
-		$type = 'Monster';
+		
+		#syntax checking for items with number in name buy whithout quotes to surrond it
+		if ($key =~ /[a-zA-Z][0-9]/ && $key !~ /"/) {
```
Comment (lututui) [inline plugins/xConf/xConf.pl]: Doesn't cover latin or chinese caracters, for example

Comment (Nipodemos) [reply plugins/xConf/xConf.pl]: solved

**Comment by Nipodemos** [comment]
nostalgia hits hard
my gosh Perl looks like a letter soup

**Pull request closed without merging**